File: 404.左叶子之和.py
#
# @lc app=leetcode.cn id=404 lang=python
#
# [404] 左叶子之和
#

# @lc code=start
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def sumOfLeftLeaves(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        self.res = 0
        def sumof(root):
            if not root:
                return 0
            if root.left and not root.left.left and not root.left.right:
                self.res += root.left.val
            sumof(root.left)
            sumof(root.right)
        sumof(root)
        return self.res
# @lc code=end

File: 404.左叶子之和.cpp
/*
 * @lc app=leetcode.cn id=404 lang=cpp
 *
 * [404] 左叶子之和
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int sum = 0;
    int sumOfLeftLeaves(TreeNode* root) {
        return helper(root);
    }
    int helper(TreeNode* root){
        if(!root)
            return 0;
        if(root->left and !root->left->left and !root->left->right)
            sum += root->left->val;
        sumOfLeftLeaves(root->left);
        sumOfLeftLeaves(root->right);
        return sum;

    }
};
// @lc code=end

